ondine/homework2.md

```diff
@@ -0,0 +1,246 @@
+### Ondine Gallatin
+
+### Homework #2
+
+Due Monday, 1/19
+
+Please also read the [learning objectives](objectives.md) for this week.
+
+---
+
+**1)** _(Easy, 10% of total time)_
+
+**a)** Write a function `fractionString(n,d)` which takes 2 parameters (n,d) and returns a string.  For example, 
+`fractionString(7,4)` should return "1 3/4", and `fractionString(3,3)` should (for now) return "1 0/3".  As before, assume that _n_ and _d_ are both positive integers.
+
+```
+function fractionString(n, d) {
+  var numeratorTwo = n % d;
+  var number = Math.floor(n / d);
+  var result = number + ' ' + numeratorTwo + '/' + d;
+  return result;
+}
+```
+
+**b)** Write a second version of your function which improves the output in two special cases:
+
+- An output string like "0 1/2" should be simplified to just "1/2";
+- An output string like "1 0/3" should be simplified to just "1".
+
+```
+function fractionStringTwo(n, d) {
+  var numeratorTwo = n % d;
+  var number = Math.floor(n / d);
+  if (number < 1) {
+    var result = numeratorTwo + '/' + d;
+  } else if (numeratorTwo == 0) {
+    result = number;
+  } else {
+    result = number + ' ' + numeratorTwo + '/' + d;
+    };
+  return result;
+}
+```
+---
+
+**2)** _(Moderate, 15%)_
+
+// not done
+
+Rewrite each block below as the simplest equivalent you can discover.  Your equivalent should produce the same final conditions as the original code given the same initial conditions.
+In each case, if there is an undeclared variable, assume it has been declared earlier and set to an unknown value.
+
+**a)**
+
+```
+var i;
+if ((n - Math.floor(n)) >= .5) {
+   i = Math.ceil(n);
+} else {
+  i = Math.floor(n);
+}
+```
+
+
+**b)**
+```
+var y, xIsFalse = (x? false : true);
+  if (xIsFalse)
+    y = false;
+  else
+   y = x;
+```
+
+```
+condition ? expr1 : expr2
+
```

Comment:
you were on the right track. What the original function says is this:

line 67) Given two variables y(undefined) and xIsFalse ( defined as the result of a conditional). if x is true xIsFalse will be FALSE, if x is FALSE, xIsFalse will become TRUE. 

line 68) if xIsFalse is true, then enter the main block. 
line 69) Y is evaluated to be the same as X. 
line70) else
line 71) if x was true, then xIsFalse will evaluate to false, and y will be come the same value of x.

essentually y will = what ever x is. and xIsFalse is possibly the worst variable name I have ever seen. It is most likely confusing you. if you rename that variable to "bool" or "val" it will probably be more clear to you. 

solution: y = (x? x: false);

this is much easier to read. y = true of x is true, and y = fase if x is false. 


and xIsFalse is whatever x is. If x is false, it is also false. if x is true it is also true. Then in the following conditional statments it does a redundant check for the value of xIsFalse. if it is true, then that means that X was false, so now make y false.